popde/density_estimate.py

```diff
@@ -74,6 +77,18 @@ def __init__(self, data, input_transf=None, stdize=False, rescale=None,
         if dim_names is not None:
             self.check_dimensionality()
 
+        self.weights = weights
+        if self.weights is not None:
+            #check the array
+            self._weights = atleast_1d(weights).astype(float)
+            #normalize weights [sum(weights)=1]
+            self._weights /= sum(self._weights)
+            if self.weights.ndim != 1:
+                raise ValueError("weights should be one-dimensional.")
+            if len(self.weights) != self.data.shape[1]:
+                raise ValueError("weights should be of length of input data")
```

Comment:
These checks should be performed before doing the normalization

Reply:
rearrange